### PR TITLE
fix: isolate macos-specific dependencies

### PR DESCRIPTION
**Pull request by aminya**
Release Notes:

- Improved isolation of macos-specific dependencies (Related to #5391 #5395 #5394)

### CODE REVIEW

**Comment by aminya** [comment]
@cla-bot check

**Comment by osiewicz** [comment]
Lgtm, except that I don't think we need a release note for that (it doesn't affect Zed binary in an user-visible manner anyhow)

File: crates/live_kit_client/Cargo.toml [hunk 1]
```diff
@@ -53,20 +55,23 @@ async-trait.workspace = true
...
-objc = "0.2"
 parking_lot.workspace = true
 serde.workspace = true
 serde_derive.workspace = true
 sha2 = "0.10"
 simplelog = "0.9"
 
+
+[target.'cfg(target_os = "macos")'.dev-dependencies]
+core-foundation = "0.9.3"
```
Comment (osiewicz) [inline crates/live_kit_client/Cargo.toml:70]: Hmm, I don't think we need `core-foundation` and `core-graphics` in `dev-dependencies`, as they're already included as normal dependencies. I know this isn't something introduced in this PR, but could we get rid of them while we're at it?

Comment (osiewicz) [reply crates/live_kit_client/Cargo.toml]: In fact, none of these dependencies are used in the `dev` context. `core-graphics` are unused too; so we can get rid of this section + remove `core-graphics` dep.

Comment (aminya) [reply crates/live_kit_client/Cargo.toml]: Sure. Fixed in [`3ab39d7` (#6854)](https://github.com/zed-industries/zed/pull/6854/commits/3ab39d7df9d88b1e327ab96fcf64367a8fce2a98)

Comment (osiewicz) [reply crates/live_kit_client/Cargo.toml]: Nice! We can also get rid of core-graphics altogether and cocoa and objc in dev-dependencies.

Comment (aminya) [reply crates/live_kit_client/Cargo.toml]: Here we go! 
[`30faa53` (#6854)](https://github.com/zed-industries/zed/pull/6854/commits/30faa538a482832364b21295d8a710584bfa7368)

Comment (osiewicz) [reply crates/live_kit_client/Cargo.toml]: Cool, thanks; I hope I'm not too nitpicky, but live-kit-client is something that has pretty big impact on our clean build release times, so whatever can be done to make it build sooner is for the better. :)

**Comment by osiewicz** [comment]
Thanks!

**Pull request merged**